qutip/solve/nonmarkov/bofin_solvers.py

```diff
@@ -376,18 +364,18 @@ def extract(self, idx_or_label):
         return Qobj(self._ado_state[idx, :].T, dims=self.rho.dims)
 
 
-class HEOMSolver:
+class HEOMSolver(Solver):
```

Comment:
It isn't. I'd like to look at how hard it would be to share more by modifying Solver slightly -- e.g. maybe `.run()` could be reused if there were some hooks to allow adding additional data to the Result, and passing the ADOs to the e_ops.

Reply:
It is now inheriting all of `.run`, `.start` and `.step` from solver -- woot!